### DISCUSSION TITLE
使用@suiet/wallet-kit 调用合约出错

### DISCUSSION THREAD

**Discussion by xiegudong45** [NonceGeek/Move-Camp#281] [Q&A] [answered]
如图，报错信息不是很详细，不知道是那个传参有问题引起的？
![image](https://user-images.githubusercontent.com/30063330/217585901-8e2820ad-5278-469e-8279-e96c9bfcbb21.png)
下图为对应的合约方法：
![image](https://user-images.githubusercontent.com/30063330/217586004-abf0d41e-eb57-4af3-a2a5-5a4a07a08a79.png)

**Comment by xiegudong45** [comment]
对比了下代码，看起来是没有传入packageObjectId.比较奇怪的是执行direnv allow好像没有反应，有谁知道是为什么吗？
![image](https://user-images.githubusercontent.com/30063330/217588514-c131f2a9-9f93-4e70-8f5f-0708ffbe9c7b.png)

**Comment by qiwihui** [comment]
请求的 “data” 应该要包含 `packageObjectId` 这个参数，比如：

```typescript
function create_example_nft() {
    const { name, url, description } = formInput;
    return {
      packageObjectId: PACKAGE_ID,
      module: 'devnet_nft',
      function: 'mint',
      typeArguments: [],
      arguments: [
        name,
        description,
        url,
      ],
      gasBudget: 30000,
    };
  }
```

**Reply by xiegudong45** [reply]
嗯是的，后面发现是direnv没生效，我执行了direnv allow好像没用，可以看下我下面的截图

**Reply by qiwihui** [reply]
贴一下生成 data 那个字段的代码吧，截图看不出来

**Comment by qiwihui** [comment, marked as answer]
使用 `.env.local` 作为环境变量配置。
